ansys/mapdl/core/logging.py

```diff
@@ -0,0 +1,499 @@
+        else:
+            raise Exception("You can only input 'str' classes to this method.")
+
+        return instance_logger
+        # return.level = None # TODO: To remove
+
+    def add_instance_logger(self, name, MAPDL_instance, level=None):
+        """
+        Create a logger for a MAPDL instance.
+        The MAPDL instance logger is a logger with an adapter which add the contextual
+        information such as MAPDL instance name. This logger is returned and you can use
+        it to log events as a normal logger. It is also stored in the ``_instances`` field.
+
+        Parameters
+        ----------
+        name : str
+            Name for the new logger
+        MAPDL_instance : ansys.mapdl.core.mapdl._MapdlCore
+            Mapdl instance object. This should contain the attribute ``name``.
+
+        Returns
+        -------
+        ansys.mapdl.core.logging.PyansysCustomAdapter
+            Logger adapter customized to add MAPDL information to the logs.
+            You can use this class to log events in the same way you would with a logger
+            class.
+
+        Raises
+        ------
+        Exception
+            You can only input strings as ``name`` to this method.
+        """
+        count_ = 0
+        new_name = name
+        while new_name in logging.root.manager.__dict__.keys():
+            count_ += 1
+            new_name = name + '_' + str(count_)
+
+        self._instances[new_name] = self._add_MAPDL_instance_logger(new_name, MAPDL_instance, level)
+        return self._instances[new_name]
+
+    def __getitem__(self, key):
+        if key in self._instances.keys():
+            return self._instances[key]
+        else:
+            raise KeyError(f"There is no instances with name {key}")
+
+    def add_handling_uncaught_expections(self, logger):
+        """This just redirect the output of an exception to the logger."""
+        def handle_exception(exc_type, exc_value, exc_traceback):
+            if issubclass(exc_type, KeyboardInterrupt):
+                sys.__excepthook__(exc_type, exc_value, exc_traceback)
+                return
+            logger.critical("Uncaught exception", exc_info=(exc_type, exc_value, exc_traceback))
+        sys.excepthook = handle_exception
+
+
+## Auxiliary functions
+
+def add_file_handler(logger, filename=FILE_NAME, level=LOG_LEVEL, write_headers=False):
+    """
+    Add a file handler to the input.
+
+    Parameters
+    ----------
+    logger : logging.Logger or logging.PyansysLogger
+        Logger where to add the file handler.
+    filename : str, optional
+        Name of the output file. By default FILE_NAME
+    level : str, optional
+        Level of log recording. By default LOG_LEVEL
+    write_headers : bool, optional
+        Record the headers to the file. By default False
+
+    Returns
+    -------
+    logger
+        Return the logger or PyansysLogger object.
+    """
+
+    _file_handler = logging.FileHandler(filename)
```

Comment:
No reason to add a leading underscore to this parameter.  Remember, parameters will be scoped to the method they are defined in; they won't leak to other methods and therefore it's fine to use a generic name like `file_handler`.